server/src/main/webapp/WEB-INF/rails/webpack/views/pages/materials/modal.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import classnames from "classnames";
+import {SparkRoutes} from "helpers/spark_routes";
+import {MithrilViewComponent} from "jsx/mithril-component";
+import _ from "lodash";
+import m from "mithril";
+import Stream from "mithril/stream";
+import {stringOrUndefined} from "models/compare/pipeline_instance_json";
+import {MaterialModification} from "models/config_repos/types";
+import {MaterialAPIs, MaterialModifications, MaterialWithFingerprint} from "models/materials/materials";
+import {FlashMessage, MessageType} from "views/components/flash_message";
+import {SearchField} from "views/components/forms/input_fields";
+import {KeyValuePair} from "views/components/key_value_pair";
+import {Link} from "views/components/link";
+import linkStyles from "views/components/link/index.scss";
+import {Modal, ModalState, Size} from "views/components/modal";
+import styles from "./index.scss";
+import {MaterialWidget} from "./material_widget";
+
+export class ShowModificationsModal extends Modal {
+  errorMessage: Stream<string>                         = Stream();
+  private material: MaterialWithFingerprint;
+  private modifications: Stream<MaterialModifications> = Stream();
+  private service: ApiService;
+  private searchQuery: Stream<string>                  = Stream("");
+
+  constructor(material: MaterialWithFingerprint, service: ApiService = new FetchHistoryService()) {
+    super(Size.large);
+    this.material = material;
+    this.service  = service;
+    this.fetchModifications();
+  }
+
+  body(): m.Children {
+    const onPatternChange = () => {
+      if (_.isEmpty(this.searchQuery()) || this.searchQuery().length < 2) {
+        return;
+      }
+      _.throttle(() => this.fetchModifications(), 500, {trailing: true})();
+    };
+
```

Comment:
Currently, the search for a modification is throttled at 500 ms. Let me know if this is not required.